include/oneapi/dpl/experimental/kt/esimd_radix_sort_utils.h

```diff
@@ -436,6 +436,100 @@ VectorStore(uint32_t offset,
+    // __uint32_vals, uint32_t, 4 bytes:                                                        V
+    // +-----------------------------+----------------------------+-----------------------------+-----------------------------+-----+-----------------------------+----------------------------+
+    // | uint32_t(d0, first 4 bytes) | uint32_t(d0, last 4 bytes) | uint32_t(d1, first 4 bytes) | uint32_t(d1, last 4 bytes)  | ... | uint32_t(dN, first 4 bytes) | uint32_t(dN, last 4 bytes) |
+    // +-----------------------------+----------------------------+-----------------------------+-----------------------------+-----+-----------------------------+----------------------------+
+
+    // mask, simd_mask<N>:
+    // +----------------------------------------------------------+-----------------------------------------------------------+-----+----------------------------------------------------------+
+    // |                             m0                           |                             m1                            | ... |                             mN                           |
+    // +----------------------------------------------------------+-----------------------------------------------------------+-----+----------------------------------------------------------+
+    //                                                                                          |
+    // __uint32_mask, simd_mask<2 * N>:                                                         V
+    // +-----------------------------+----------------------------+-----------------------------+-----------------------------+-----+-----------------------------+----------------------------+
+    // |             m0              |             m0             |             m1              |             m1              | ... |             mN              |             mN             |
+    // +-----------------------------+----------------------------+-----------------------------+-----------------------------+-----+-----------------------------+----------------------------+
+
+    using TypeX32 = decltype(DestTypeResolver(T{}));
+
+    // https://intel.github.io/llvm-docs/doxygen/classsycl_1_1__V1_1_1ext_1_1intel_1_1esimd_1_1detail_1_1simd__obj__impl.html#ad1eaca7ca43fcf87a29a9714b93d8109
+    // bit_cast_view
+    sycl::ext::intel::esimd::simd<TypeX32, 2 * N> __x32_vals = vals.template bit_cast_view<TypeX32>();
+
+    __ESIMD_NS::simd<uint32_t, 2 * N> __x32_offsets;
+    __ESIMD_NS::simd_mask<2 * N>      __x32_mask(0);
+
+    // TODO required to implement this through simd/mask operations
+    // simd select stride
+    #pragma unroll
+    for (size_t src_index = 0; src_index < N; ++src_index)
+    {
+        __x32_offsets[2 * src_index    ] = offsets[src_index];
+        __x32_offsets[2 * src_index + 1] = offsets[src_index] + sizeof(TypeX32);
+
+        __x32_mask[2 * src_index    ] = mask[src_index];
+        __x32_mask[2 * src_index + 1] = mask[src_index];
+    }
```

Comment:
Have you tried using `select`?
```suggestion
    __x32_offsets.template select<N, 2>(0) = offsets[src_index];
    __x32_offsets.template select<N, 2>(1) = offsets[src_index] + sizeof(TypeX32);
    __x32_mask.template<select<N, 2>(0) = mask[src_index];
    __x32_mask.template<select<N, 2>(1) = mask[src_index];
```

Reply:
This new function was removed.